x/gov/keeper/vote.go

```diff
@@ -19,6 +21,11 @@ func (k Keeper) AddVote(ctx sdk.Context, proposalID uint64, voterAddr sdk.AccAdd
 	if !ok {
 		return sdkerrors.Wrapf(govtypes.ErrUnknownProposal, "%v", proposalID)
 	}
+
+	if voterAddr.String() == "" {
+		return status.Error(codes.InvalidArgument, "invalid request")
```

Comment:
I don't think we need this because `MsgVote` type has a method `ValidateBasic()` that ensures a non-empty voter field. If we keep this, we would need to return a custom error like in line 22.

Reply:
Okay, the reason I added this check was that I wrote a [test](https://github.com/priyanshiiit/shentu/blob/c1e8d2dfa67e2ebf17130569318724e79b63e989/x/gov/keeper/grpc_query_test.go#L483-L496) for empty address and it panicked. Should I remove this test?

Reply:
I think panicking is fine here.